rayon-core/src/registry.rs

```diff
@@ -602,17 +600,8 @@ impl WorkerThread {
         if num_threads <= 1 {
             return None;
         }
-        assert!(num_threads < (u32::MAX as usize),
-                "we do not support more than u32::MAX worker threads");
-
-        let start = {
-            // OK to use this UnsafeCell because (a) this data is
-            // confined to current thread, as WorkerThread is not Send
-            // nor Sync and (b) rand crate will not call back into
-            // this method.
-            let rng = &mut *self.rng.get();
-            rng.next_u32() % num_threads as u32
-        } as usize;
+
+        let start = self.rng.next() as usize % num_threads;
```

Comment:
Nit: since we control `rng`, why not add a nice wrapper function like `self.rng.next_usize(num_threads)`  that hides all this modulo stuff?